src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -210,4 +239,60 @@ private static string GetUserAgentString()
             return UserAgentProduct;
         }
     }
+
+    private void ApplyConfiguration(
+        IConfiguration configuration,
+        OtlpExporterOptionsConfigurationType configurationType)
+    {
+        Debug.Assert(configuration != null, "configuration was null");
+
+        // Note: When using the "AddOtlpExporter" extensions configurationType
+        // never has a value other than "Default" because OtlpExporterOptions is
+        // shared by all signals and there is no way to differentiate which
+        // signal is being constructed.
+        if (configurationType == OtlpExporterOptionsConfigurationType.Default)
+        {
+            this.ApplyConfigurationUsingSpecificationEnvVars(
```

Comment:
Shouldn't this applied always?. What if user provides default environment variables for few options and signal specific for other options?

Reply:
Good question! I was wondering if anyone would spot that 🤣 It does seem like a mistake, but it is on purpose!

Keep in mind this is for the cross-cutting extension (#5400). I don't have a solution for using the signal-specific envvars with the current extensions. I'm thinking they will only be supported via the cross-cutting one.

What the cross-cutting extension does is build up an options class like this...

```csharp
class OtlpExporterBuilderOptions
{
    public IOtlpExporterOptions DefaultOptions { get; } // Gets env vars from Default keys

    public IOtlpExporterOptions LoggingOptions { get; } // Gets env vars from Log keys
    public IOtlpExporterOptions MetricsOptions { get; } // Gets env vars from Metric keys
    public IOtlpExporterOptions TracingOptions { get; } // Gets env vars from Trace keys
}
```

When it goes to build an exporter for say metrics it does...

```csharp
options.MetricsOptions.ApplyDefaults(options.DefaultOptions)
```

What that does is take values from `DefaultOptions` for anything NOT set on `MetricsOptions`.

That gives us the behavior the spec defines w.r.t. envvars, but also allows users to set things on defaults which will also apply.

Reply:
got it - sounds good to me. 

> I'm thinking they will only be supported via the cross-cutting one

Just to confirm, we should still be able to add support for signal specific env var in future for `AddOtlpExporter`, if we need to?

Reply:
The issue is Options API. All signals use the same `OtlpExporterOptions`. What we _need_ is a way in the ctor to know which signal is being constructed so we can lookup/apply the correct envvars. If each signal had its own class, it would be easy. On this PR the ctor is invoked with a parameter. The cross-cutting extension is able to do that but I haven't found an elegant way to do that yet for the existing extensions. If we can find some way to figure out what signal is being managed then yes we could tweak this logic and support it. I have tried a couple things that sort of work, but nothing so far that wasn't a hack/had edge cases where it wouldn't work.